### PR TITLE
Add modifying of VAST for video bids and add validation

### PR DESCRIPTION
**Pull request by DGarbar**
- Add validation for Video bids. `bid.adm` or `bid.nurl` needs to be present
  - This case now, is not possible `<VASTAdTagURI><![CDATA[null]]></VASTAdTagURI>`
- Bid adm will be updated as cache. (see https://github.com/prebid/prebid-server/issues/1015)
- Bid type is defined by bidder, not our (`imp.video != null`, etc) checks. For example Appnexus use `bid.ext.appnexus.bid_ad_type` to define it. (that's why there are a lot of changed cache jsons. Also add ordering for tests)

Refactored a bit.
- Removed confusing maps
- Removed confusing checks
- Removed several imp to bid matching

Possible improvements (in another ticket bc current PR is too large)
- Extract more event URL to another class
- Use bidInfo in BidResponseCreator for BidResponse

### CODE REVIEW

File: src/main/java/org/prebid/server/auction/BidResponseCreator.java [hunk 1]
```diff
@@ -243,6 +255,37 @@ private static ExtRequestTargeting targeting(BidRequest bidRequest) {
...
+                .filter(Objects::nonNull)
+                .map(BidderSeatBid::getBids)
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .map(bidderBid -> toBidInfo(bidderBid.getBid(), bidderBid.getType(), imps, bidderResponse.getBidder()))
+                .collect(Collectors.toList());
+    }
+
+    private static BidInfo toBidInfo(Bid bid, BidType type, List<Imp> imps, String bidder) {
+        final Imp imp = correspondingImp(bid, imps);
```
Comment (schernysh) [inline src/main/java/org/prebid/server/auction/BidResponseCreator.java:270]: This variable could be inlined

File: src/main/java/org/prebid/server/auction/BidResponseCreator.java [hunk 2]
```diff
@@ -384,98 +428,60 @@ private static void tryAddWinningBidByBidder(Bid bid, String bidder,
...
+                                                 EventsContext eventsContext) {
         if (!cacheInfo.isDoCaching()) {
             return Future.succeededFuture(CacheServiceResult.of(null, null, toMapBidsWithEmptyCacheIds(bidsToCache)));
         }
 
         // do not submit non deals bids with zero price to prebid cache
-        final List<Bid> bidsValidToBeCached = bidsToCache.stream()
+        final List<BidInfo> bidsValidToBeCached = bidsToCache.stream()
                 .filter(BidResponseCreator::isValidForCaching)
+                // for test consistency
```
Comment (schernysh) [inline src/main/java/org/prebid/server/auction/BidResponseCreator.java:445]: If this isn't required for production usage, I would recommend removing this step.

File: src/main/java/org/prebid/server/vast/VastModifier.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+import java.util.Set;
+
+public class VastModifier {
+
+    private final BidderCatalog bidderCatalog;
+    private final EventsService eventsService;
+
+    public VastModifier(BidderCatalog bidderCatalog, EventsService eventsService) {
+        this.bidderCatalog = bidderCatalog;
```
Comment (schernysh) [inline src/main/java/org/prebid/server/vast/VastModifier.java:20]: Minor: please add null checks

File: src/test/resources/org/prebid/server/it/openrtb2/rubicon_appnexus/test-auction-rubicon-appnexus-response.json [hunk 4]
```diff
@@ -267,7 +267,7 @@
           "id": "880290288",
           "impid": "impId1",
           "price": 8.43,
-          "adm": "<Impression><![CDATA[]]><\/Impression>",
+          "adm": "<Impression><![CDATA[]]></Impression><Impression><![CDATA[http://localhost:8080/event?t=imp&b=880290288&a=5001&ts=1000&bidder=rubicon&f=b&int=dmbjs]]></Impression>",
```
Comment (schernysh) [inline src/test/resources/org/prebid/server/it/openrtb2/rubicon_appnexus/test-auction-rubicon-appnexus-response.json:270]: Perhaps `localhost:8080` part is better to be replaced with a placeholder. Please consider this improvement in other modified test resources (where applicable).

File: src/main/java/org/prebid/server/auction/BidResponseCreator.java (hunk 2 above)
Comment (DGarbar) [reply src/main/java/org/prebid/server/auction/BidResponseCreator.java:445]: I couldn't get permanent /cache request order otherwise.

Comment (schernysh) [reply src/main/java/org/prebid/server/auction/BidResponseCreator.java:444]: Please take a look at https://github.com/prebid/prebid-server-java/blob/3ce7df94f7b6f42a4d4eaa9fb52d10b54f048b93/src/test/java/org/prebid/server/it/IntegrationTest.java#L185. Does this help or at least gives an idea how to make tests reliable?

**Review by rpanchyk — APPROVED** [review]

**Pull request merged**